openedx/core/djangoapps/credit/tasks.py

```diff
@@ -75,8 +74,16 @@ def _get_course_credit_requirements(course_key):
     credit_xblock_requirements = _get_credit_course_requirement_xblocks(course_key)
     min_grade_requirement = _get_min_grade_requirement(course_key)
     proctored_exams_requirements = _get_proctoring_requirements(course_key)
+    block_requirements = credit_xblock_requirements + proctored_exams_requirements
+    try:
+        sorted_block_requirements = sorted(block_requirements, key=lambda x: (x['start_date'], x['display_name']))
+    except TypeError:
+        # some of the blocks don't have start date set on them so sort blocks
+        # on just the 'display_name' field
+        sorted_block_requirements = sorted(block_requirements, key=lambda x: (x['display_name']))
```

Comment:
This seems like it could lead to some unexpected results.  If I'm understanding this right, this means that if any requirement doesn't have a start date, then the blocks will be sorted by display name (and blocks with the same name will be secondarily sorted by start date).  I can see this being really confusing for course teams.

I think it would make more sense to put requirements with no start date at the beginning of the list.  Setting the start date to `None` in Studio means, "This problem is released when the course opens", so I think that would make more sense to course authors.  We can still do a secondary sort on display name in case two requirements have the same start date.

A cheap way to achieve this would be to:
1) Set `start_date` to the minimum datetime (`datetime.MINYEAR`) if the XBlock has a `start_date` set to `None`.
2) Use the sorting logic on line 79, but remove lines 80-83.